doit.go

```diff
@@ -40,26 +46,123 @@ func (c *LiveConfig) GetGodoClient() *godo.Client {
 	return godo.NewClient(oauthClient)
 }
 
+func sshConnect(user string, host string, method ssh.AuthMethod) (err error) {
+	sshc := &ssh.ClientConfig{
+		User: user,
+		Auth: []ssh.AuthMethod{method},
+	}
+	conn, err := ssh.Dial("tcp", host, sshc)
+	if err != nil {
+		return err
+	}
+
+	session, err := conn.NewSession()
+	if err != nil {
+		return err
+	}
+
+	session.Stdout = os.Stdout
+	session.Stderr = os.Stderr
+	session.Stdin = os.Stdin
+
+	defer session.Close()
+
+	modes := ssh.TerminalModes{
+		ssh.ECHO: 1,
+	}
+
+	var (
+		termWidth, termHeight int
+	)
+	fd := os.Stdin.Fd()
+	if term.IsTerminal(fd) {
+		oldState, err := term.MakeRaw(fd)
+		if err != nil {
+			return err
+		}
+
+		defer term.RestoreTerminal(fd, oldState)
+
+		winsize, err := term.GetWinsize(fd)
+		if err != nil {
+			termWidth = 80
+			termHeight = 24
+		} else {
+			termWidth = int(winsize.Width)
+			termHeight = int(winsize.Height)
+		}
+	}
+
+	if err := session.RequestPty("xterm", termWidth, termHeight, modes); err != nil {
+		session.Close()
+		return err
+	}
+	if err == nil {
+		err = session.Shell()
+	}
+	if err != nil {
+		return err
+	}
+
+	err = session.Wait()
+	if err != nil && err != io.EOF {
+		// Ignore the error if it's an ExitError with an empty message,
+		// this occurs when you do CTRL+c and then run exit cmd which isn't an
+		// actual error.
+		waitMsg, ok := err.(*ssh.ExitError)
+		if ok && waitMsg.Msg() == "" {
+			return nil
+		}
+
+		return err
+	}
+	return err
+}
+
 // SSH creates a ssh connection to a host.
-func (c *LiveConfig) SSH(user, host string, options []string) Runner {
+func (c *LiveConfig) SSH(user, host, keyPath string) (err error) {
```

Comment:
Wrapping this in a type called `Runner` with a single function called `Run` seems like an unnecessary abstraction. Especially when this was the only place in the repo that you do that.

How do you feel about removing SSH from LiveConfig and putting it all in ssh.go?